modules/core/src/main/java/org/apache/synapse/rest/RESTRequestHandler.java

```diff
@@ -159,15 +156,25 @@ private boolean identifyAPI(API api, MessageContext synCtx, List defaultStrategy
                     log.debug("Located specific API: " + api.getName() + " for processing message");
                 }
                 apiProcessNonDefaultStrategy(synCtx, api);
-                return true;
+                return new States(defaultAPI, true);
             }
         } else if (api.canProcess(synCtx)) {
             if (log.isDebugEnabled()) {
                 log.debug("Located specific API: " + api.getName() + " for processing message");
             }
             api.process(synCtx);
-            return true;
+            return new States(defaultAPI, true);
         }
-        return false;
+        return new States(defaultAPI, false);
+    }
+}
+
+class States {
+    API defaultAPI;
+    boolean state;
```

Comment:
state = true/false is misleading. e.g., I don't understand the meaning of `if (states.state)' at line 89. Can we have a better wording?

Reply:
Shall we go with 'isApiProcessed' instead of state?

Reply:
Resolving this as we have modified the fix.